catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -0,0 +1,116 @@
+ * metacard updates and forwards changes on to any publish locations.
+ */
+public class RegistryPublicationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPublicationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
+
+    private final FederationAdminService adminService;
+
+    private final ExecutorService executor;
+
+    public RegistryPublicationHandler(FederationAdminService adminService,
+            ExecutorService service) {
+        this.adminService = adminService;
+        this.executor = service;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+        try {
+            if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                    LOGGER.error("Thread pool didn't terminate");
+                }
+            }
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+        }
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+        executor.execute(() -> processUpdate(mcard));
+    }
+
+    private void processUpdate(Metacard mcard) {
+        Attribute publishedLocations =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (publishedLocations == null) {
+            return;
+        }
+
+        Attribute lastPublished = mcard.getAttribute(RegistryObjectMetacardType.LAST_PUBLISHED);
+        Date datePublished = null;
+        if (lastPublished != null) {
+            datePublished = (Date) lastPublished.getValue();
+        }
+
+        Set<String> locations = publishedLocations.getValues()
+                .stream()
+                .map(Object::toString)
+                .collect(Collectors.toCollection(HashSet::new));
+
+        //If we haven't published this metacard before or if we have had an update
+        //since the last time we published send an update to the remote location
+        if ((datePublished == null || datePublished.before(mcard.getModifiedDate()))
+                && CollectionUtils.isNotEmpty(locations)) {
+            try {
+                adminService.updateRegistryEntry(mcard, locations);
+                mcard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.LAST_PUBLISHED,
+                        mcard.getModifiedDate()));
+                adminService.updateRegistryEntry(mcard);
+            } catch (FederationAdminException e) {
+                LOGGER.warn("Unable to send update for {} to {}", mcard.getTitle(), locations);
```

Comment:
If this happened, what could an administrator do to send the update once the federation issue is resolved?